File: app/src/main/java/com/example/ase_project_nutrismart/Response/PurchasedGrocery.java
package com.example.ase_project_nutrismart.Response;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.io.Serializable;
import java.util.ArrayList;

@Entity
public class PurchasedGrocery implements Serializable {
    @PrimaryKey(autoGenerate = true)
    private int id;

    public ArrayList<Grocery> getData() {
        return data;
    }

    public void setData(ArrayList<Grocery> data) {
        this.data = data;
    }

    @ColumnInfo(name="data")
    public ArrayList<Grocery> data;


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }


}

File: app/src/main/java/com/example/ase_project_nutrismart/Response/PurchasedGrocery.kt
package com.example.ase_project_nutrismart.Response

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.ase_project_nutrismart.Response.BmiData
import com.example.ase_project_nutrismart.Response.ExpireData
import com.example.ase_project_nutrismart.Response.Grocery
import com.example.ase_project_nutrismart.Response.RolesResponse
import java.io.Serializable
import java.util.ArrayList

@Entity
class PurchasedGrocery : Serializable {
    @PrimaryKey(autoGenerate = true)
    var id = 0

    @JvmField
    @ColumnInfo(name = "data")
    var data: ArrayList<Grocery>? = null
}